File: flink-java-core/src/main/java/com/atguigu/flink/day3/RichFunctionExample.java
package com.atguigu.flink.day3;

import org.apache.flink.api.common.functions.RichMapFunction;
import org.apache.flink.configuration.Configuration;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

/**
 * @author ：hmz
 * @date ：Created in 2020/9/12 11:45
 */
public class RichFunctionExample {

    public static void main(String[] args) throws  Exception{
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

        env.setParallelism(1);
        env.fromElements(1,2,3,4)
                .map(new RichMapFunction<Integer, Integer>() {
                    @Override
                    public void open(Configuration parameters) throws Exception {
                        System.out.println("open");
                    }

                    @Override
                    public Integer map(Integer value) throws Exception {
                        return value+1;
                    }

                    @Override
                    public void close() throws Exception {
                        System.out.println("close");
                    }
                })
                .print();


        env.execute();
    }
}

File: flink-scala-core/src/main/scala/com/atguigu/flink/day3/RichFunctionExample.scala
package com.atguigu.flink.day3

import org.apache.flink.api.common.functions.RichMapFunction
import org.apache.flink.configuration.Configuration
import org.apache.flink.streaming.api.scala._

object RichFunctionExample {

    def main(args: Array[String]): Unit = {
        val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
        env.setParallelism(1)

        val stream: DataStream[Int] = env.fromElements(1, 2, 3)
        stream.map(new MyRichMapFunctrion).print()

        env.execute();
    }

    class MyRichMapFunctrion extends RichMapFunction[Int,Int]{

        override def open(parameters: Configuration): Unit = {
            println("open")
        }

        override def map(value: Int): Int = {
            value+1
        }

        override def close(): Unit = {
            println("close")
        }
    }

}
